src/3p.js

```diff
@@ -109,18 +109,25 @@ function executeAfterWriteScript(win, fn) {
  * @param {string} src
  */
 export function validateSrcPrefix(prefix, src) {
-
+  const validPrefixes = [];
   if (!isArray(prefix)) {
     prefix = [prefix];
   }
-
-  for (const p of prefix) {
-    if (src.indexOf(p) === 0) {
+  const totalPrefixes = prefix.length;
+  if (src !== undefined) {
+    let p = 0;
+    for (p = 0; p <= totalPrefixes; p++) {
+      const protocolIndex = src.toLowerCase().indexOf(prefix[p]);
+      if (protocolIndex == 0) {
+        validPrefixes.push(p.toString());
```

Comment:
The intent here is "one of" semantics. It is sufficient that one prefix matches.